presto-base-arrow-flight/src/main/java/com/facebook/plugin/arrow/ArrowConnector.java

```diff
@@ -0,0 +1,77 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.plugin.arrow;
+
+import com.facebook.presto.spi.connector.Connector;
+import com.facebook.presto.spi.connector.ConnectorMetadata;
+import com.facebook.presto.spi.connector.ConnectorPageSourceProvider;
+import com.facebook.presto.spi.connector.ConnectorSplitManager;
+import com.facebook.presto.spi.connector.ConnectorTransactionHandle;
+import com.facebook.presto.spi.transaction.IsolationLevel;
+import com.google.inject.Inject;
+import org.apache.arrow.memory.BufferAllocator;
+
+import static java.util.Objects.requireNonNull;
+
+public class ArrowConnector
+        implements Connector
+{
+    private final ConnectorMetadata metadata;
+    private final ConnectorSplitManager splitManager;
+    private final ConnectorPageSourceProvider pageSourceProvider;
+    private final BufferAllocator connectorAllocator;
+
+    @Inject
+    public ArrowConnector(
+            ConnectorMetadata metadata,
+            ConnectorSplitManager splitManager,
+            ConnectorPageSourceProvider pageSourceProvider,
+            BufferAllocator connectorAllocator)
+    {
+        this.metadata = requireNonNull(metadata, "Metadata is null");
+        this.splitManager = requireNonNull(splitManager, "splitManager is null");
+        this.pageSourceProvider = requireNonNull(pageSourceProvider, "pageSourceProvider is null");
+        this.connectorAllocator = requireNonNull(connectorAllocator, "connectorAllocator is null");
```

Comment:
Changed this to use a BufferAllocator instance. The connector will own this allocator and close it on shutdown. Flight clients created by the connector create a child allocator from this that is closed with the client.